pandas/core/array_algos/masked_reductions.py

```diff
@@ -57,6 +58,8 @@ def _reductions(
         else:
             return func(values, axis=axis, **kwargs)
     else:
+        mask |= isna(values)
```

Comment:
As I checked, the issue occurs only if isinstance(delegate, ExtensionArray) case of [Series::_reduce](https://github.com/pandas-dev/pandas/blob/main/pandas/core/series.py#L6588).
Despite the skipna is passed to [_reductions](https://github.com/pandas-dev/pandas/blob/main/pandas/core/array_algos/masked_reductions.py#L59), it is not applied to mask.

So, to apply skipna=True, the 'isna' need to be applied to 'mask'.
Other types like 'int' do not come into this flow since they don't use ExtensionArray

Reply:
Below is my understanding.  Please let me know if it was wrong.
① Normal case :  No issue.  Not affected by this PR.
② Issue case : There was  no code to deal with 'skipna'.  This PR need to be applied for this flow.
![image](https://github.com/user-attachments/assets/06265a9a-b239-467d-bc59-50ef3509843d)

Reply:
Oh, I see it, thank you.  
As I checked again, both 'Float64' and 'Int64' works well in normal situation.

pd.Series([pd.NA,1.,2.],dtype='Float64').mean(skipna=True)
 --> No problem. BaseMaskedArray::_mask was set [ True False False] as intended.
 
However, In the issue's case below,
series3's BaseMaskedArray::_mask was initialized to [ False False False] ,
then didn't modified to [ True False False] despite the 1st data became <NA> after div.


series1 = pd.Series({"a": 0.0, "b": 1, "c": 1})
series2 = pd.Series({"a": 0.0, "b": 2, "c": 2})
series3 = series1.convert_dtypes() / series2.convert_dtypes()
series3.mean(skipna=True)


So, now I'd like to suggest to apply this mask operation after div in BaseMaskedArray::_arith_method.

Reply:
When `skipna=True`, we should skip NA values regardless of how they get into the `values`. Doing this in in `_arith_method` does not accomplish this. I think this PR was closer with the previous implementation.

In addition, changing the mask as this PR is doing in `truediv` is not a change we should make lightly. I do not think we should be touching this logic in this PR.

Reply:
OK. I  reverted to the previous patch, and reduced the scope to np.flat64